workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLifecycleOwner.kt

```diff
@@ -111,9 +111,9 @@ internal class RealWorkflowLifecycleOwner(
   LifecycleEventObserver {
 
   /**
-   * This gets nulled out when we are destroyed.
+   * Weak reference ensures that we don't leak the view.
    */
-  private var view: View? = view
+  private var view: WeakReference<View> = WeakReference(null)
```

Comment:
👀 would love to learn more about what "be even more careful" means. How comes we couldn't clear out the view ref when we needed to?

https://square.github.io/leakcanary/fundamentals-fixing-a-memory-leak/
![image](https://user-images.githubusercontent.com/557033/133696513-a5c0c167-a473-4162-b115-0ae79809d27c.png)

Reply:
For more context: this PR is part of the not-quite-finished effort to make Workflow get along with Compose. 

* Compose relies on SavedStateRegistry instead of classic View persistence (saveHierarchyState et al)
* SavedStateRegistry reliies on LifecycleOwner
* We need to set up LifecyleOwner + SavedStateRegistry per backstack frame and pre dialog
* The timing for all of that is hard to get right

Full details are in #458, which I'm going to have to internalize better.

Reply:
#560 gets rid of the `WeakReference` with no apparent ill effects.